app/src/main/java/com/intive/selftraining/selftraining/movieDetails/MovieRepository.kt

```diff
@@ -1,16 +1,31 @@
 package com.intive.selftraining.selftraining.movieDetails
 
+import com.intive.selftraining.selftraining.data.MovieDetailsMapper
 import com.intive.selftraining.selftraining.movieDetails.model.MovieDetails
+import com.intive.selftraining.selftraining.network.models.movieDetails.MovieDetailsEntitiy
 import com.intive.selftraining.selftraining.network.NetworkInterface
+import com.intive.selftraining.selftraining.network.models.listMovies.ConfigurationEntity
 import io.reactivex.Observable
 import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.rxkotlin.Observables
 import io.reactivex.schedulers.Schedulers
 
 class MovieRepository(private val networkClient: NetworkInterface) {
 
     fun getMovieResult(id: Int): Observable<MovieDetails> {
+        return Observables.zip(getMovie(id), getConfiguration()) {
+                movie, configuration ->
+            MovieDetailsMapper().mapFromEntity(movie, configuration.images)
+        }
+    }
+
+    private fun getMovie(id: Int): Observable<MovieDetailsEntitiy> {
```

Comment:
maybe it could be named showMovieDetails, just like in the ListMoviesRepository, for the sake of a pattern in the naming? Let me know what you think.